subsys/bluetooth/host/att.c

```diff
@@ -2948,6 +2989,61 @@ static int bt_att_accept(struct bt_conn *conn, struct bt_l2cap_chan **ch)
 BT_L2CAP_CHANNEL_DEFINE(att_fixed_chan, BT_L2CAP_CID_ATT, bt_att_accept, NULL);
 
 #if defined(CONFIG_BT_EATT)
+static k_timeout_t credit_based_connection_delay(struct bt_conn *conn)
+{
+	/*
+	 * 5.3 Vol 3, Part G, Section 5.4 L2CAP COLLISION MITIGATION
+	 * ... In this situation, the Central may retry
+	 * immediately but the Peripheral shall wait a minimum of 100 ms before retrying;
+	 * on LE connections, the Peripheral shall wait at least 2 *
+	 * (connPeripheralLatency + 1) * connInterval if that is longer.
+	 */
+
+	if (IS_ENABLED(CONFIG_BT_CENTRAL) && conn->role == BT_CONN_ROLE_CENTRAL) {
+		return K_NO_WAIT;
+
+	} else if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+		const uint32_t rand_delay = sys_rand32_get() & 0x7; /* Small random delay for IOP */
```

Comment:
Why not use ‘bt_rand()‘ instead as that’s the normal way of getting random data within the Bluetooth subsystem? Then you wouldn’t need to select any additional Kconfig symbols either.

Reply:
`ENTROPY_GENERATOR` is already selected by `BT_LL_SW_SPLIT` and `BT_CTRL_CRYPTO`, so it is already selected when running on target in most cases. The config is only missing for the shell application when compiling for native.

Reply:
@hermabe This PR is for the host, not the controller, so I'm not completely convinced. In fact, you didn't address my main question: what's wrong with using `bt_rand()` when that's what the Bluetooth host stack uses for getting random data in all other places? Respectively, if we are to use some other way of getting random data then that other way should be used everywhere in the host - not just in this one place.

Reply:
`bt_rand()` seems to work as well. Can switch to that to be consistent with the rest of the host.